articles/server-apis/aspnet-core-webapi/02-authorization.md

```diff
@@ -0,0 +1,136 @@
+
+![Configure Client](/media/articles/server-apis/aspnet-core-webapi/configure-api-client-scopes.png)
+
+## 2. Enforcing scopes in your ASP.NET Core API 
+
+To ensure that a correct `scope` is present in order to execute a particular API endpoint, you can make use of the new Policy Based Authorization in ASP.NET Core. 
+
+::: panel-info ASP.NET Core Policy Based Authorization
+For a better understanding of the code which follows, it is suggested that you read the ASP.NET Core documentation on [Policy Based Authorization](https://docs.microsoft.com/en-us/aspnet/core/security/authorization/policies).
+:::
+
+Create a new Authorization Requirement called `HasScopeRequirement`. This requirement will check that the `scope` claim issued by your Auth0 tenant is present, and if so it will ensure that the `scope` claim contains the requested scope. If it does then the Authorization Requirement is met. 
+
+```csharp
+public class HasScopeRequirement : AuthorizationHandler<HasScopeRequirement>, IAuthorizationRequirement
+{
+    private readonly string issuer;
+    private readonly string scope;
+
+    public HasScopeRequirement(string scope, string issuer)
+    {
+        this.scope = scope;
+        this.issuer = issuer;
+    }
+
+    protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, HasScopeRequirement requirement)
+    {
+        // If user does not have the scope claim, get out of here
+        if (!context.User.HasClaim(c => c.Type == "scope" && c.Issuer == issuer))
+            return Task.CompletedTask;
+
+        // Split the scopes string into an array
+        var scopes = context.User.FindFirst(c => c.Type == "scope" && c.Issuer == issuer).Value.Split(' ');
+
+        // Succeed if the scope array contains the required scope
+        if (scopes.Any(s => s == scope))
+            context.Succeed(requirement);
+
+        return Task.CompletedTask;
+    }
+}
+``` 
+
+Next you can define a policy for each of the scopes in your application in the `ConfigureServices` method of your `Startup` class: 
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    // Add framework services.
+    services.AddMvc();
+
+    string domain = $"https://{Configuration["Auth0:Domain"]}/";
+    services.AddAuthorization(options =>
+    {
+        options.AddPolicy("read:timesheets",
+            policy => policy.Requirements.Add(new HasScopeRequirement("read:timesheets", domain)));
+        options.AddPolicy("create:timesheets",
+            policy => policy.Requirements.Add(new HasScopeRequirement("create:timesheets", domain)));
+    });
+}
+```
+
+Finally, to ensure that a scope is present in order to call a particular API endpoint, you simply need to decorate the action with the `Authorize` attribute, and pass the name of the Policy for that `scope` in the `policy` parameter:
+
+```csharp
+[Route("api/timesheets")]
+public class TimesheetsController : Controller
+{
+    [Authorize("read:timesheets")]
```

Comment:
Not necessary. The first parameter is `policy`:

![image](https://cloud.githubusercontent.com/assets/1006420/20627324/d0baa794-b351-11e6-82a5-5ca7e4b52f67.png)